web/src/SidebarAccount.tsx

```diff
@@ -0,0 +1,268 @@
+  }
+`
+let SidebarAccountMenuHeader = styled.header`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  border-bottom: 1px dotted ${Color.grayLight};
+  padding-top: ${SizeUnit(0.25)};
+  padding-bottom: ${SizeUnit(0.25)};
+  padding-left: ${SizeUnit(0.5)};
+  padding-right: ${SizeUnit(0.5)};
+`
+let SidebarAccountMenuLogo = styled(TiltCloudLogoSvg)`
+  fill: ${Color.text};
+`
+let SidebarAccountMenuLearn = styled.a`
+  font-family: ${Font.sansSerif};
+  font-size: ${FontSize.smallest};
+  color: ${Color.grayLight};
+`
+let SidebarAccountMenuContent = styled.div`
+  font-family: ${Font.monospace};
+  font-size: ${FontSize.default};
+  color: ${Color.grayLight};
+  padding: ${SizeUnit(0.5)};
+
+  p + p {
+    margin-top: ${SizeUnit(0.25)};
+  }
+
+  strong {
+    font-weight: bold;
+    color: ${Color.text};
+  }
+  small {
+    font-size: ${FontSize.small};
+  }
+  &.is-signedIn {
+    margin-top: ${SizeUnit(0.3)};
+    text-align: right;
+  }
+  &.is-signedOut {
+  }
+`
+let MenuContentTeamInTiltfile = styled.small`
+  opacity: 0;
+  margin-top: 0;
+  display: block;
+  transition: opacity ${AnimDuration.default} ease;
+  transition-delay: ${AnimDuration.short};
+`
+let MenuContentTeamName = styled.strong`
+  transition: background-color ${AnimDuration.default} ease;
+  transition-delay: ${AnimDuration.short};
+`
+let MenuContentTeam = styled.p`
+  &:hover {
+    ${MenuContentTeamInTiltfile} {
+      opacity: 1;
+    }
+    ${MenuContentTeamName} {
+      background-color: ${Color.offWhite};
+    }
+  }
+`
+let MenuContentSignInLink = styled.p`
+  text-align: center;
+`
+export const MenuContentButtonTiltCloud = styled(ButtonLink)`
+  margin-top: ${SizeUnit(0.3)};
+`
+export const MenuContentButtonSignUp = styled(ButtonInput)`
+  margin-top: ${SizeUnit(0.5)};
+  margin-bottom: ${SizeUnit(0.25)};
+`
+
+type SidebarAccountProps = {
+  tiltCloudUsername: string | null
+  tiltCloudSchemeHost: string
+  tiltCloudTeamID: string | null
+}
+
+function notifyTiltOfRegistration() {
+  let url = `${window.location.protocol}//${window.location.host}/api/user_started_tilt_cloud_registration`
+  fetch(url, {
+    method: "POST",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+  })
+}
+
+function SidebarAccount(props: SidebarAccountProps) {
+  const [accountMenuIsVisible, toggleAccountMenu] = useState(false)
+  return (
+    <SidebarAccountRoot>
+      <SidebarAccountHeader>
+        <SidebarAccountButton
+          onClick={() => toggleAccountMenu(!accountMenuIsVisible)}
+        >
+          <SidebarAccountLabel>Your Tilt Cloud status</SidebarAccountLabel>
+          <SidebarAccountIcon />
+        </SidebarAccountButton>
+      </SidebarAccountHeader>
+      <SidebarAccountMenu className={accountMenuIsVisible ? "isVisible" : ""}>
+        <SidebarAccountMenuHeader>
+          <SidebarAccountMenuLogo></SidebarAccountMenuLogo>
+          {!props.tiltCloudUsername && (
```

Comment:
i feel like we should pull the conditionals out?

having them inline like this makes this very hard to read.

e.g.,

```
let optionalLearnMore = null
if (!props.tiltCloudUsername) {
  optionalLearnMore = ...
}